crates/sui-move-build/src/linters/freeze_wrapped.rs

```diff
@@ -105,152 +105,55 @@ pub struct FreezeWrappedVisitor {
 }
 
 impl TypingVisitor for FreezeWrappedVisitor {
-    fn visit(
-        &mut self,
-        env: &mut CompilationEnv,
-        _program_info: &ProgramInfo,
-        program: &mut T::Program,
-    ) {
-        for (_, _, mdef) in program.modules.iter() {
-            env.add_warning_filter_scope(mdef.warning_filter.clone());
-
-            for (_, _, fdef) in mdef.functions.iter() {
-                env.add_warning_filter_scope(fdef.warning_filter.clone());
-
-                if let T::FunctionBody_::Defined(seq) = &fdef.body.value {
-                    self.visit_seq(seq, env, &program.modules);
-                }
-
-                env.pop_warning_filter_scope();
-            }
-
-            env.pop_warning_filter_scope();
-        }
-    }
-}
-
-impl FreezeWrappedVisitor {
-    fn visit_seq_item(
-        &mut self,
-        sp!(_, seq_item): &T::SequenceItem,
-        env: &mut CompilationEnv,
-        modules: &UniqueMap<E::ModuleIdent, T::ModuleDefinition>,
-    ) {
-        use T::SequenceItem_ as SI;
-        match seq_item {
-            SI::Seq(e) => self.visit_exp(e, env, modules),
-            SI::Declare(_) => (),
-            SI::Bind(_, _, e) => self.visit_exp(e, env, modules),
-        }
-    }
-
-    fn visit_exp(
+    fn visit_exp_custom(
         &mut self,
         exp: &T::Exp,
         env: &mut CompilationEnv,
-        modules: &UniqueMap<E::ModuleIdent, T::ModuleDefinition>,
-    ) {
+        _program_info: &ProgramInfo,
+        program: &T::Program,
+    ) -> bool {
         use T::UnannotatedExp_ as E;
-        let sp!(_, uexp) = &exp.exp;
-        match uexp {
-            E::ModuleCall(fun) => {
-                if FREEZE_FUNCTIONS.iter().any(|(addr, module, fname)| {
-                    fun.module.value.is(*addr, *module) && &fun.name.value().as_str() == fname
-                }) {
-                    let Some(bt) = base_type(&fun.type_arguments[0]) else {
+        if let E::ModuleCall(fun) = &exp.exp.value {
+            if FREEZE_FUNCTIONS.iter().any(|(addr, module, fname)| {
+                fun.module.value.is(*addr, *module) && &fun.name.value().as_str() == fname
+            }) {
+                let Some(bt) = base_type(&fun.type_arguments[0]) else {
                         // not an (potentially dereferenced) N::Type_::Apply nor N::Type_::Param
-                        return;
+                        return true;
                     };
-                    let N::Type_::Apply(_,tname, _) = &bt.value else {
+                let N::Type_::Apply(_,tname, _) = &bt.value else {
                         // not a struct type
-                        return;
+                        return true;
                     };
```

Comment:
Yeah, it did not occur to me that someone could use a "void" function in any meaningful way in an argument, but better be safe than sorry. 

I guess we have to process arguments in  ALL cases, though, even if the diag gets generated, so `visit_exp_custom` should return false no matter what. 

Will fix it once https://github.com/MystenLabs/sui/pull/13335 lands in case you need to make some more visitor changes so that I don't have to deal with conflicts.